docs/source/app-dev/bindings-java/code-snippets/ProductTypes.daml

```diff
@@ -0,0 +1,10 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+-- start snippet: product types example
+daml 1.2
+module Com.Acme where
+
+data Person = Person with name : Name; age : Decimal
+data Name = Name with firstName : Text; lastName : Text
+-- end snippet: product types example
```

Comment:
I recommend configuring your editor to add trailing newlines.

Reply:
fixed!